dataprotocol-client/src/protocolConverter.ts

```diff
@@ -45,6 +47,152 @@ function asProviderMetadata(params: types.MetadataQueryResult): data.ProviderMet
+			databaseInfoOptions: new Array<data.ServiceOption>(),
+			databaseFileInfoOptions: new Array<data.ServiceOption>(),
+			fileGroupInfoOptions: new Array<data.ServiceOption>()
+		};
+
+		if (result.capabilities.adminServicesProvider.databaseInfoOptions
+			&& result.capabilities.adminServicesProvider.databaseInfoOptions.length > 0) {
+			for (let i = 0; i < result.capabilities.adminServicesProvider.databaseInfoOptions.length; ++i) {
+				let srcOption: any = result.capabilities.adminServicesProvider.databaseInfoOptions[i];
+				let descOption: data.ServiceOption = buildServiceOption(srcOption);
+				capabilities.adminServicesProvider.databaseInfoOptions.push(descOption);
+			}
+		}
+
+		if (result.capabilities.adminServicesProvider.databaseFileInfoOptions
+			&& result.capabilities.adminServicesProvider.databaseFileInfoOptions.length > 0) {
+			for (let i = 0; i < result.capabilities.adminServicesProvider.databaseFileInfoOptions.length; ++i) {
+				//let srcOption: types.ServiceOption = result.capabilities.adminServicesProvider.databaseFileInfoOptions[i];
+				let srcOption: any = result.capabilities.adminServicesProvider.databaseFileInfoOptions[i];
+				let descOption: data.ServiceOption = buildServiceOption(srcOption);
+				capabilities.adminServicesProvider.databaseFileInfoOptions.push(descOption);
+			}
+		}
+
+		if (result.capabilities.adminServicesProvider.fileGroupInfoOptions
+			&& result.capabilities.adminServicesProvider.fileGroupInfoOptions.length > 0) {
+			for (let i = 0; i < result.capabilities.adminServicesProvider.fileGroupInfoOptions.length; ++i) {
+				//let srcOption: types.ServiceOption = result.capabilities.adminServicesProvider.fileGroupInfoOptions[i];
+				let srcOption: any = result.capabilities.adminServicesProvider.fileGroupInfoOptions[i];
+				let descOption: data.ServiceOption = buildServiceOption(srcOption);
+				capabilities.adminServicesProvider.fileGroupInfoOptions.push(descOption);
+			}
+		}
+	}
+
+	if (result.capabilities.connectionProvider
+		&& result.capabilities.connectionProvider.options
+		&& result.capabilities.connectionProvider.options.length > 0) {
+		capabilities.connectionProvider = <data.ConnectionProviderOptions>{
+			options: new Array<data.ConnectionOption>()
+		};
+		for (let i = 0; i < result.capabilities.connectionProvider.options.length; ++i) {
+			let srcOption: any = result.capabilities.connectionProvider.options[i];
+			let descOption: data.ConnectionOption = {
+				name: srcOption.name,
+				displayName: srcOption.displayName ? srcOption.displayName : srcOption.name,
+				description: srcOption.description,
+				groupName: srcOption.groupName,
+				defaultValue: srcOption.defaultValue,
+				categoryValues: srcOption.categoryValues,
+				isIdentity: srcOption.isIdentity,
+				isRequired: srcOption.isRequired,
+				valueType: asServiceOptionType(srcOption.valueType),
+				specialValueType: undefined
+			};
+
+			if (srcOption.specialValueType === 'serverName') {
```

Comment:
I remember doing this now. I removed this because it doesn't make sense to convert a string to a number just to compare it to a enum later. we can just make the enum a string enum and compare them directly. There must have been something I overlooked when I made this change, but it still should be true.

Reply:
@anthonydresser in general I agree with your point.  Unfortunately the way the change was made brakes the Advanced Connection Dialog.